File: T3H/Buoi14_animation/app/src/main/java/com/example/buoi14/MainActivity.java
package com.example.buoi14;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import com.example.buoi14.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity implements MainListener {
    private ActivityMainBinding binding;
    private Animation alpha;
    private Animation rotate;
    private Animation translate;
    private Animation scale;
    private Animation set;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setListener(this);
        initAnimation();
    }

    private void initAnimation() {
        alpha = AnimationUtils.loadAnimation(this, R.anim.alpha);
        rotate = AnimationUtils.loadAnimation(this, R.anim.rotate);
        translate = AnimationUtils.loadAnimation(this, R.anim.translate);
        scale = AnimationUtils.loadAnimation(this, R.anim.scale);
        set = AnimationUtils.loadAnimation(this, R.anim.set);
    }

    @Override
    public void onAlphaClicked(View view) {
        view.startAnimation(alpha);
    }

    @Override
    public void onRotateClicked(View view) {
        view.startAnimation(rotate);
    }

    @Override
    public void onTranslateClicked(View view) {
        view.startAnimation(translate);
    }

    @Override
    public void onScaleClicked(View view) {
        view.startAnimation(scale);
    }

    @Override
    public void onSetClicked(View view) {
        view.startAnimation(set);
    }
}

File: Personal/DatabaseAndCallApi/app/src/main/java/com/vanphuc/databaseandcallapi/MainActivity.kt
package com.vanphuc.databaseandcallapi

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.vanphuc.databaseandcallapi.data.model.Photo
import com.vanphuc.databaseandcallapi.data.resource.remote.PhotoApi
import com.vanphuc.databaseandcallapi.databinding.ActivityMainBinding
import com.vanphuc.databaseandcallapi.ui.base.BaseRecyclerview
import retrofit2.*
import retrofit2.converter.gson.GsonConverterFactory


class MainActivity : AppCompatActivity() {
    private var binding: ActivityMainBinding? = null
    private var service: PhotoApi? = null
    private val recyclerView by lazy {
        BaseRecyclerview<Photo>(
            R.layout.photo_item
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        val retrofit = Retrofit.Builder()
            .baseUrl("https://jsonplaceholder.typicode.com")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
        service = retrofit.create(PhotoApi::class.java)
        binding!!.apply {
            recyclerPhoto.adapter = recyclerView
        }
        binding?.btnLoad?.setOnClickListener {
            service?.listPhoto()?.enqueue(object : Callback<List<Photo>> {
                override fun onResponse(
                    call: Call<List<Photo>>,
                    response: Response<List<Photo>>
                ) {
                    response.body()?.let { it1 -> recyclerView.updateData(it1) }
                }

                override fun onFailure(call: Call<List<Photo>>, t: Throwable) {

                }
            })
        }
    }
}